### PR TITLE
Update to Core beta.23

### PR DESCRIPTION
**Pull request by julianrex**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

## Pull request checklist:
 - [X] Briefly describe the changes in this PR.
 - [X] Document any changes to public APIs.
 - [X] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog></changelog>`.
 - [ ] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes

This PR:
- updates to MapboxCoreMaps 10.0.0-beta.22, MapboxCommon 13.0.0, MapboxMobileEvents 1.0.0
- Removes the Cartfile, Carthage directory and associated `deps` target from the makefile.
- `MBXFeature` has been renamed `Feature`, `MBXGeometry` renamed as `Geometry`.  Existing uses of the `Turf` types have been fully namespaced. 
- Removes the top-level `MapboxMaps.xcworkspace`, as it's no longer required.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -449,10 +449,6 @@ ifndef SDK_REGISTRY_TOKEN
 endif
 	@echo "$$NETRC" > $(NETRC_FILE)
 
-.PHONY: deps
-deps: | $(NETRC_FILE)
-	XCODE_XCCONFIG_FILE=${PWD}/xcode-12.xcconfig carthage bootstrap --platform iOS --use-netrc --no-cache-builds
```
Comment (neelmistry94) [inline Makefile:454]: can we get rid of this `Xcode-12.xcconfig` file now?

Comment (neelmistry94) [reply Makefile:454]: Nvm, you did, it's the last file

**Review by neelmistry94 — CHANGES_REQUESTED** [review]
This is quite a large change to drop Carthage so I'd request the following
- scrub our CONTRIBUTING.md and DEVELOPING.md to remove any Carthage instructions
- Add a changelog bullet that development environment is now on SPM and not carthage
- Also, in our base.xcconfig, let's reenable warnings as errors. I have tested this and our builds do not fail anymore

File: Makefile (hunk 1 above)
Comment (julianrex) [reply Makefile:454]: It should be removed as part of this PR.

File: DEVELOPING.md [hunk 2]
```diff
@@ -8,11 +8,17 @@ This project runs on Apple's Metal rendering framework. Prerequisites for runnin
...
-This project requires [a valid ~/.netrc file](https://docs.mapbox.com/ios/maps/guides/install/#configure-credentials) to fetch dependencies.
+This project:
 
-This project uses [SwiftLint](https://github.com/realm/SwiftLint) to enforce a style guide. Install it with `brew install swiftlint` and check for lint by running `swiftlint` in the root of the repository.
+1. requires [a valid ~/.netrc file](https://docs.mapbox.com/ios/maps/guides/install/#configure-credentials) to fetch dependencies.
 
-This project uses [secret-shield](https://github.com/mapbox/secret-shield) to help block secrets such as access tokens from being exposed. Install `secret-shield` by entering `npm install -g @mapbox/secret-shield`. Install the pre-commit hook by running `scripts/install-pre-commit/install-pre-commit.sh`
+1. uses [SwiftLint](https://github.com/realm/SwiftLint) to enforce a style guide. Install it with `brew install swiftlint` and check for lint by running `swiftlint` in the root of the repository.
+
+1. uses [Swift Package Manager](https://github.com/apple/swift-package-manager) to manage dependencies for development. Cocapods is supported for *consuming* the SDK.
```
Comment (neelmistry94) [inline DEVELOPING.md:17]: the numbers also 1

**Review by neelmistry94 — APPROVED** [review]

File: DEVELOPING.md (hunk 2 above)
Comment (julianrex) [reply DEVELOPING.md:17]: This is fine for Markdown, but does look odd when viewed in a text editor. Will update.

**Pull request merged**